table-print/Sources/TPTablePrint.swift

```diff
@@ -83,28 +93,64 @@ open class TPTablePrint {
         let numberOfExtraKeySpace = (widths.key - pair.key.count)
         let extraKeySpaces = repeatChar(" ", count: numberOfExtraKeySpace)
         let valueString = pair.value
-        let numberOfExtraValueSpace = (widths.value - valueString.count)
-        let extraValueSpaces = repeatChar(" ", count: numberOfExtraValueSpace)
         let paddingSpace = repeatChar(" ", count: configuration.padding)
-        let line = configuration.verticalLineSymbol
+
+        var result = configuration.verticalLineSymbol
             + paddingSpace
             + pair.key
             + extraKeySpaces
             + paddingSpace
             + configuration.verticalLineSymbol
             + paddingSpace
-            + valueString
-            + extraValueSpaces
+
+        let lineEnd = paddingSpace + configuration.verticalLineSymbol
+            + "\n"
+            + configuration.verticalLineSymbol
+            + repeatChar(" ", count: widths.key + 2 * configuration.padding)
+            + configuration.verticalLineSymbol
             + paddingSpace
+
+        result += breakLines(valueString, lineLength: widths.value, lineEnd: lineEnd)
+
+        return result
+    }
+
+    private func breakLines(_ string: String, lineLength: Int, lineEnd: String) -> String {
+        var result = ""
+
+        var rest = Substring(string)
+        var currentLineLength = 0
+        while !rest.isEmpty {
+            let breakIndex = rest.firstIndex { $0.isWhitespace } ?? rest.endIndex
+            let count = rest.distance(from: rest.startIndex, to: breakIndex)
+
+            if currentLineLength == 0 && count > lineLength {
+                result += rest.prefix(lineLength)
+                result += lineEnd
+
+                rest = rest.dropFirst(lineLength)
+            } else if currentLineLength + count > lineLength {
+                result += repeatChar(" ", count: lineLength - currentLineLength)
+                result += lineEnd
+
+                currentLineLength = 0
+            } else {
+                currentLineLength += count + 1
+                result += rest.prefix(upTo: breakIndex)
+                result += " "
+                rest = rest.suffix(from: breakIndex)
+                rest = rest.drop(while: { $0.isWhitespace })
+            }
+        }
+
+        result += repeatChar(" ", count: configuration.padding + lineLength - currentLineLength)
             + configuration.verticalLineSymbol
-        return line
+
+        return result
+
```

Comment:
Looks great. It's just a personal preference, but I don't like blank lines in between statements. In my opinion they don't help understanding the functionality. I'd rather break up the method into multiple smaller methods. I admit though, that this is hard and also that the code is hard to read anyway. What do you think?